qiskit_nature/circuit/library/ansatzes/ucc.py

```diff
@@ -247,6 +248,8 @@ def _build(self) -> None:
         if self._data is not None:
             return
 
+        self.operators: Optional[Union[OperatorBase, QuantumCircuit, list]]
```

Comment:
I assume this typehint is here as it lacking on the EvolvedOperatorAnsatz in Terra. (similarly in the other changed file)

Reply:
Yes. Should I fix this in Terra?

Reply:
In my mind, since the Terra version we derive from defines operators and what the types are I think the typehint should be there. That module does have typehints on some methods, and not on others, such as operators. And in that regard it is different than what is here which looks like hints were removed in places - @Cryoris any comment/thought on this?

Reply:
We currently can't put the typehint in the circuit library due to circular imports between the circuit library and opflow